File: Middleware/docs/examples/code/HttpGetAsync.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class HttpGetAsync {
	
	public static void main(String[] args) throws Exception {
		String runnerId = "d593c930-7fed-4c7b-ac52-fff946b78c32";
		String requestUrl = "http://localhost:8082/apply/async/" + runnerId;

		URL url = new URL(requestUrl);
		HttpURLConnection connection = (HttpURLConnection) url.openConnection();
		connection.setRequestMethod("GET");

		int responseCode = connection.getResponseCode();
		System.out.println("GET Response Code: " + responseCode);

		try (BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream(), "utf-8"))) {
			StringBuilder response = new StringBuilder();
			String responseLine;
			while ((responseLine = br.readLine()) != null) {
				response.append(responseLine.trim());
			}
			System.out.println(response.toString());
		}
	}
}

File: Middleware/docs/examples/code/HttpGetAsync.js
async function httpGetAsync() {
    const runnerId = "d593c930-7fed-4c7b-ac52-fff946b78c32";
    const requestUrl = `http://localhost:8082/apply/async/${runnerId}`;

    try {
        const response = await fetch(requestUrl);
        console.log("GET Response Code:", response.status);
        const data = await response.text();
        console.log(data);
    } catch (error) {
        console.error("Error:", error);
    }
}

httpGetAsync();